firmware-binaries/examples/smoltcp_echo/memory.x

```diff
@@ -6,8 +6,8 @@ SPDX-License-Identifier: CC0-1.0
 
 MEMORY
 {
-  IMEM : ORIGIN = 0x80000000, LENGTH = 256K
-  DMEM : ORIGIN = 0x10000000, LENGTH = 64K
+  IMEM : ORIGIN = 0x80000000, LENGTH = 1024K
+  DMEM : ORIGIN = 0x10000000, LENGTH = 256K
```

Comment:
Do you really need 4 times more memory? We already use a lot IMO.

Reply:
We already have 4 times more memory, because currently we define
1024k and 256k "elements" of `BitVector 32`:
https://github.com/bittide/bittide-hardware/blob/staging/bittide/src/Bittide/DoubleBufferedRam.hs#L42-L45
https://github.com/bittide/bittide-hardware/blob/staging/bittide/src/Bittide/DoubleBufferedRam.hs#L190

It's a bit misleading, but not trivial to fix since we do need `Initialcontents d (Bytes 4)`